src/sentry/sentry_metrics/consumers/indexer/schema_validator.py

```diff
@@ -0,0 +1,69 @@
+import random
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+from sentry_kafka_schemas.codecs import Codec, ValidationError
+
+from sentry import options
+from sentry.sentry_metrics.consumers.indexer.parsed_message import ParsedMessage
+
+
+class MetricsSchemaValidator(ABC):
+    @abstractmethod
+    def validate(self, message: ParsedMessage) -> None:
+        """
+        This method tells whether a message is valid or not. It should raise a
+        ValidationError if the message is not valid. Else it should return None.
+        """
+        pass
+
+
+class GenericMetricsSchemaValidator(MetricsSchemaValidator):
```

Comment:
Couldn't release health use the exact same class, with a sample rate set to 1.0? I'm not sure I understand the necessity of two classes for this.

Reply:
Currently there are a few implicit differences between the generic metrics pipeline and the release health pipeline. The primary purpose of having 2 separate classes was to make things more explicit. 

Having said that, I made a single class with the option name configurable. That can be used to differentiate the pipeline.